File: frontend/console/src/api/ai.js
import Axios from "@/http";

export function createAi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/ai', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveAi(aiId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/ai/${ aiId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateAi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/ai', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteAi(aiId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/ai/${ aiId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteAiWithFlag(aiId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/ai/${ aiId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllAis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/ai/all').then(res => {
            resolve(res)
        })
    })
}

export function queryAi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/ai/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/ai.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_AI = "/config/ai"
const URL_ROUTER_AI_BY_ID = "/config/ai/:ai-id"

func AiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_AI, config.CreateAi)
	r.GET(URL_ROUTER_AI_BY_ID, config.RetrieveAi)
	r.PUT(URL_ROUTER_AI, config.UpdateAi)
	r.DELETE(URL_ROUTER_AI_BY_ID, config.DeleteAi)
	r.PUT(URL_ROUTER_AI_BY_ID, config.DeleteAiWithFlag)
	return r
}
